### ISSUE TITLE
About Pre-Training Single Task Checkpoint and Data Preprocessing File bfp_dataset_script.py

### ISSUE THREAD

**Issue by kleinercubs** [antonio-mastropaolo/TransferLearning4Code#4]
Hi! The topic of your paper is quite interesting! I'm trying to reproduce the result of your paper. I'd like to double check with you that whether Pre-Training Single Task model the same one for the performance that show in your paper? 
It seems that I can't reach similar results as the paper shown. Can you share some details about the data preprocessing in `bfp_dataset_script.py`

**Comment by antonio-mastropaolo** [comment]
Hello Li,

Could you be more specific about the problem you're having?
For example, which is the task and so on

**Comment by kleinercubs** [comment]
Hi Antonio,

I'm doing the experiment on raw assertion genearation task with the pretraining single model. Currently, I only got 58 for top1 accuracy. However, I think it should be around 68 for top1 according to the results shown in the paper?

**Comment by antonio-mastropaolo** [comment]
Gotcha,

So for starter, you should use the following script (https://drive.google.com/file/d/16CZavq_dhT3TmUtRuFIYDUAwygNHTb6a/view?usp=sharing) when generating prediction for the AG-task.

Second, make sure that when you use the script to generate predictions (https://github.com/antonio-mastropaolo/TransferLearning4Code/blob/main/Code/Run-on-test/Generate-Result.ipynb), cell #5 and #6 are aligned with the task of Assert Statement Generation. In other words, you should have the following code for cell #5 and #6 respectively

```
#!cp /content/drive/MyDrive/T5-Extension/Miscellaneous/bfp_dataset_script.py /content/bfp_dataset_script.py
!cp /content/drive/MyDrive/T5-Extension/Miscellaneous/assert_dataset_script.py /content/assert_dataset_script.py
```

```
# Change the prefix when want to evaluate different tasks
# (1) generate small patch
# (2) generate medium patch
# (3) generate abt assert
# (4) generate raw assert

# If you're evaluating abt/raw assert generative tasks, change example['method'].lower() for the input_text and  example['assertion'].lower() for the target_text

def add_eos_to_examples(example):
    example['input_text'] = 'generate raw assert: %s ' % example['method']#.lower()
    example['target_text'] = '%s ' % example['assertion']#.lower()
    return example


def convert_to_features(example_batch):
    input_encodings = tokenizer.batch_encode_plus(example_batch['input_text'], pad_to_max_length=True, max_length=512)
    target_encodings = tokenizer.batch_encode_plus(example_batch['target_text'], pad_to_max_length=True, max_length=512)

    encodings = {
        'input_ids': input_encodings['input_ids'], 
        'attention_mask': input_encodings['attention_mask'],
        'target_ids': target_encodings['input_ids'],
        'target_attention_mask': target_encodings['attention_mask']
    }

    return encodings
```

Similarly, when generating prediction for the abstracted code the prefix has to be:
"generate abs assert"

I hope this clarifies your doubts.

**Comment by kleinercubs** [comment]
Thanks! This is really helpful!
